gpjax/abstractions.py

```diff
@@ -125,42 +130,56 @@ def fit_batches(
     trainables: tp.Dict,
     train_data: Dataset,
     optax_optim,
+    seed: tp.Union[int, PRNGKey],
+    batch_size: int,
     n_iters: tp.Optional[int] = 100,
     log_rate: tp.Optional[int] = 10,
-) -> tp.Dict:
+) -> tp.Tuple[tp.Dict, f64["n_iters"]]:
     """Abstracted method for fitting a GP model with mini-batches respect to a supplied objective function.
     Optimisers used here should originate from Optax.
     Args:
         objective (tp.Callable): The objective function that we are optimising with respect to.
         params (dict): The parameters for which we would like to minimise our objective function with.
         trainables (dict): Boolean dictionary of same structure as 'params' that determines which parameters should be trained.
         train_data (Dataset): The training dataset.
-        opt_init (tp.Callable): The supplied optimiser's initialisation function.
-        opt_update (tp.Callable): Optimiser's update method.
-        get_params (tp.Callable): Return the current parameter state set from the optimiser.
+        optax_optim (GradientTransformation): The Optax optimiser that is to be used for learning a parameter set.
+        seed (int): The random seed for the mini-batch sampling.
+        batch_size(int): The batch_size.
         n_iters (int, optional): The number of optimisation steps to run. Defaults to 100.
         log_rate (int, optional): How frequently the objective function's value should be printed. Defaults to 10.
     Returns:
-        tp.Dict: An optimised set of parameters.
+        tp.Tuple[tp.Dict, f64["n_iters"]]: A tuple comprising optimised parameters and training history respectively.
     """
 
     opt_state = optax_optim.init(params)
-    next_batch = train_data.get_batcher()
+
+    prng = convert_seed(seed)
```

Comment:
Is there a specific reason for this? Why can't we just require the user to supply a Jax PRNGKey?

Reply:
The thinking behind this was to match how things are done in distrax. With fresh eyes, I would agree that this is superfluous. Lets remove it.